.gitmodules

```diff
@@ -1,3 +1,6 @@
 [submodule "third-party/googletest"]
 	path = third-party/googletest
 	url = https://github.com/google/googletest.git
+[submodule "third_party/libuv"]
```

Comment:
I understand that it will be the default transport implementation on Windows, but it's already the default transport implementation on macOS and we didn't make it github submodule, so why do we need to make it github submodule now? do we really need to make it github submodule? maybe it makes sense to expect it as an installed dependency? cc @mrshenli

Reply:
Since in PyTorch libuv is a sub module of tensorpipe, and for windows, just enable gloo, so I made it as a sub module of gloo.  My thinking is if both tensorpipe and gloo enabled when building PyTorch, there is only one libuv target can be created, so libuv will not be built twice. I've not tried to install libuv on windows, let me try it. 
Do you mean that on macOS by default gloo will use libuv?
@mrshenli ,

Reply:
> Do you mean that on macOS by default gloo will use libuv?

Yes, on OSX, Gloo will use libuv, and we currenct expect OSX users to install pkg-config and libuv through conda. 

Just checked TensorPipe did include libuv as a submodule, and recall there was some issue with that during build. 

cc @beauby @malfet Since both TensorPipe and Gloo will use libuv, what's the recommended way here? should we pull libuv as a submodule into PyTorch, or should we keep them separate?

Reply:
@beauby @malfet I will work on find_package on Windows for Libuv, will update the it in this PR.

Reply:
@gunandrose4u: For Windows I suggest using [`find_library()`](https://cmake.org/cmake/help/latest/command/find_library.html), and relying on the user to provide `libuv_ROOT` to CMake. Similarly, you could use `find_file()` for the headers.

Reply:
Sure, let me figure out and switch to find_library().